.github/workflows/ruby.yml

```diff
@@ -0,0 +1,20 @@
+name: Run Specs
+on: [push, pull_request]
+
+jobs:
+  test_embedded_ruby:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        ruby: [ '2.6', '2.7', '3.0', '3.1', '3.2', '3.3', '3.4', head]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v6
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - run: bundle install
```

Comment:
Running bundle install here masks a compatibility issue. The bundle install pulls in a newer json gem (≥2.5.0) which provides JSON.load_file, but this method doesn't exist in Ruby's stdlib json library for versions < 3.0.